mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMessageMapper.java

```diff
@@ -239,25 +240,21 @@ public MessageUid findFirstUnseenMessageUid(Mailbox mailbox) throws MailboxExcep
             case FROM:
                 return findMessagesInMailboxBetweenUIDs(mailbox, MaildirMessageName.FILTER_DELETED_MESSAGES, from, null, -1);
             case ALL:
-            default:
                 return findMessagesInMailbox(mailbox, MaildirMessageName.FILTER_DELETED_MESSAGES, -1);
+            default:
+                throw new RuntimeException();
```

Comment:
can you add at least a description?

Reply:
https://github.com/linagora/james-project/pull/2189